### PR TITLE
feat(resources): add categories to Resource Status

### PR DESCRIPTION
**Pull request by adr-mo**
## Description

This is the final pull request to integrate categories to Resource Status.
New capabilities

- Category chip added to the detail panel
- Filtering capabilities added on categories `service|host_category_names`

https://www.loom.com/share/8e2bccb9c26946d6ae460e40af569f9b

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.04.x
- [ ] 21.10.x
- [ ] 22.04.x
- [x] 22.10.x (master)

<h2> How this pull request can be tested ? </h2>

See jira ticket for details for validation

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: doc/API/centreon-api-v22.10.yaml [hunk 1]
```diff
@@ -294,6 +296,31 @@ paths:
           $ref: '#/components/responses/Forbidden'
         '500':
           $ref: '#/components/responses/InternalServerError'
+  /monitoring/services/categories:
+    get:
+      tags:
+        - Service category
+      summary: "List all service categories"
+      description: |
+        List of all service categories in real time context
```
Comment (cg-tw) [inline doc/API/centreon-api-v22.10.yaml]:
```suggestion
        List of all service categories in real-time context
```

File: doc/API/centreon-api-v22.10.yaml [hunk 2]
```diff
@@ -1553,6 +1582,31 @@ paths:
           $ref: '#/components/responses/Forbidden'
         '500':
           $ref: '#/components/responses/InternalServerError'
+  /monitoring/hosts/categories:
+    get:
+      tags:
+        - Host category
+      summary: "List all host categories"
+      description: |
+        List of all host categories in real time context
```
Comment (cg-tw) [inline doc/API/centreon-api-v22.10.yaml]:
```suggestion
        List of all host categories in real-time context
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 3]
```diff
@@ -16917,3 +16917,18 @@ msgstr "Alias parent"
 
 msgid "The following bound attributes are missing: %s"
 msgstr "Les attributs liés suivants sont manquants : %s"
+
+msgid "An error occured while retrieving service categories"
+msgstr "Une erreur s'est produite lors de la récupération des catégories de service"
```
Comment (cg-tw) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgstr "Une erreur s'est produite lors de la récupération des catégories de services"
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 4]
```diff
@@ -16917,3 +16917,18 @@ msgstr "Alias parent"
...
+
+msgid "An error occured while retrieving host categories"
+msgstr "Une erreur s'est produite lors de la récupération des catégories d'hôte"
```
Comment (cg-tw) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgstr "Une erreur s'est produite lors de la récupération des catégories d'hôtes"
```

File: src/Core/Category/RealTime/Application/UseCase/FindHostCategory/FindHostCategory.php [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+     * @param FindHostCategoryPresenterInterface $presenter
+     */
+    public function __invoke(FindHostCategoryPresenterInterface $presenter): void
+    {
+        $this->info('Searching for host categories');
+
+        try {
+            $hostCategories = $this->repository->findAllByTypeId(Tag::HOST_CATEGORY_TYPE_ID);
+        } catch (\Throwable $e) {
+            $this->error('An error occured while retrieving host categories');
```
Comment (cg-tw) [inline src/Core/Category/RealTime/Application/UseCase/FindHostCategory/FindHostCategory.php]:
```suggestion
            $this->error('An error occurred while retrieving host categories');
```

File: src/Core/Category/RealTime/Application/UseCase/FindServiceCategory/FindServiceCategory.php [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+     * @param FindServiceCategoryPresenterInterface $presenter
+     */
+    public function __invoke(FindServiceCategoryPresenterInterface $presenter): void
+    {
+        $this->info('Searching for service categories');
+
+        try {
+            $serviceCategories = $this->repository->findAllByTypeId(Tag::SERVICE_CATEGORY_TYPE_ID);
+        } catch (\Throwable $e) {
+            $this->error('An error occured while retrieving service categories');
```
Comment (cg-tw) [inline src/Core/Category/RealTime/Application/UseCase/FindServiceCategory/FindServiceCategory.php]:
```suggestion
            $this->error('An error occurred while retrieving service categories');
```

File: src/Core/Category/RealTime/Application/UseCase/FindServiceCategory/FindServiceCategory.php [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+            $presenter->setResponseStatus(new ErrorResponse('An error occured while retrieving service categories'));
```
Comment (cg-tw) [inline src/Core/Category/RealTime/Application/UseCase/FindServiceCategory/FindServiceCategory.php]:
```suggestion
            $presenter->setResponseStatus(new ErrorResponse('An error occurred while retrieving service categories'));
```

File: tests/php/Core/Application/RealTime/UseCase/FindMetaService/FindMetaServiceTest.php [hunk 8]
```diff
@@ -42,313 +40,222 @@
...
+    $presenter = new FindMetaServicePresenter($this->hypermediaCreator, $this->presenterFormatter);
+    $findMetaService(1, $presenter);
+
+    expect($presenter->getResponseStatus())->toBeInstanceOf(NotFoundResponse::class);
+    expect($presenter->getResponseStatus()?->getMessage())->toBe(
+        'MetaService configuration not found'
+    );
+});
+
+it('should present a NotFoundResponse if meta service configuraiton not found as non-admin', function () {
```
Comment (cg-tw) [inline tests/php/Core/Application/RealTime/UseCase/FindMetaService/FindMetaServiceTest.php]:
```suggestion
it('should present a NotFoundResponse if meta service configuration not found as non-admin', function () {
```

File: tests/php/Core/Category/RealTime/Application/UseCase/FindHostCategory/FindHostCategoryTest.php [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+        ->willThrowException(new \Exception());
+
+    $useCase = new FindHostCategory($this->tagRepository);
+
+    $presenter = new FindHostCategoryPresenterStub();
+    $useCase($presenter);
+
+    expect($presenter->getResponseStatus())->toBeInstanceOf(ErrorResponse::class);
+    expect($presenter->getResponseStatus()?->getMessage())->toBe(
+        'An error occured while retrieving host categories'
```
Comment (cg-tw) [inline tests/php/Core/Category/RealTime/Application/UseCase/FindHostCategory/FindHostCategoryTest.php]:
```suggestion
        'An error occurred while retrieving host categories'
```

File: tests/php/Core/Category/RealTime/Application/UseCase/FindServiceCategory/FindServiceCategoryTest.php [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+        ->willThrowException(new \Exception());
+
+        $useCase = new FindServiceCategory($repository);
+
+    $presenter = new FindServiceCategoryPresenterStub();
+    $useCase($presenter);
+
+    expect($presenter->getResponseStatus())->toBeInstanceOf(ErrorResponse::class);
+    expect($presenter->getResponseStatus()?->getMessage())->toBe(
+        'An error occured while retrieving service categories'
```
Comment (cg-tw) [inline tests/php/Core/Category/RealTime/Application/UseCase/FindServiceCategory/FindServiceCategoryTest.php]:
```suggestion
        'An error occurred while retrieving service categories'
```

**Review by jeremyjaouen — DISMISSED** [review]

**Review by jeremyjaouen — DISMISSED** [review]

**Review by kduret — DISMISSED** [review]

File: www/front_src/src/Resources/Details/models.ts [hunk 11]
```diff
@@ -15,12 +15,16 @@ export interface Group extends NamedEntity {
   configuration_uri: string | null;
 }
 
+export interface Category extends NamedEntity {
+  configuration_uri: string | null;
+}
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/models.ts:20]:
```suggestion
}

```

**Review by callapa — DISMISSED** [review]

**Review by callapa — DISMISSED** [review]

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 12]
```diff
@@ -16917,3 +16917,9 @@ msgstr "Alias parent"
 
 msgid "The following bound attributes are missing: %s"
 msgstr "Les attributs liés suivants sont manquants : %s"
+
+msgid "An error occured while retrieving service categories"
+msgstr "Une erreur s'est produite lors de la récupération des catégories de services"
+
+msgid "An error occured while retrieving host categories"
+msgstr "Une erreur s'est produite lors de la récupération des catégories d'hôtes"
```
Comment (Thebarda) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:16925]:
```suggestion
msgstr "Une erreur s'est produite lors de la récupération des catégories d'hôtes"

```

File: www/front_src/src/Resources/Filter/index.test.tsx [hunk 13]
```diff
@@ -170,6 +170,7 @@ const filter = {
       object_type: 'service_groups',
       value: [webAccessServiceGroup],
     },
+
```
Comment (Thebarda) [inline www/front_src/src/Resources/Filter/index.test.tsx]:
```suggestion
```

**Review by Thebarda — COMMENTED** [review]
Don't forget to add translation for spanish and portuguese

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po (hunk 12 above)
Comment (adr-mo) [reply lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:16925]: Done ✅

**Review by Thebarda — APPROVED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Review by bdauria — APPROVED** [review]

**Review by callapa — APPROVED** [review]

**Review by jdelpierre — APPROVED** [review]

**Review by cg-tw — APPROVED** [review]

**Pull request merged**